### PR TITLE
Introduce storybook into development process

### PR DESCRIPTION
**Pull request by danielleroux**
## 🆕 What is the new behavior?

Introduce storybook

(/cc #1130)

## 🏁 Checklist

A pull request can only be merged if all of these conditions are met (where applicable):

- [ ] 🦮 Accessibility (a11y) features were implemented
- [ ] 🗺️ Internationalization (i18n) - no hard coded strings
- [ ] 📲 Responsiveness - components handle viewport changes and content overflow gracefully
- [ ] 📄 Documentation was reviewed/updated (`pnpm run docs`)
- [ ] 🧪 Unit tests were added/updated and pass (`pnpm test`)
- [ ] 📸 Visual regression tests were added/updated and pass ([Guide](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing))
- [ ] 🧐 Static code analysis passes (`pnpm lint`)
- [x] 🏗️ Successful compilation (`pnpm build`, changes pushed)

### CODE REVIEW

File: packages/storybook-docs/src/stories/Button.stories.ts
Comment (jul-lam) [inline packages/storybook-docs/src/stories/Button.stories.ts:1]: Filename sollte lower case sein.

File: packages/storybook-docs/src/stories/Modal.stories.ts
Comment (jul-lam) [inline packages/storybook-docs/src/stories/Modal.stories.ts:1]: Filename sollte lower case sein.

File: packages/core/package.json [hunk 1]
```diff
@@ -29,12 +29,11 @@
...
-    "build:scss": "ts-node -P ./scripts/build/tsconfig.json ./scripts/build/build-css.ts",
-    "build:dev": "stencil build",
+    "build": "stencil build --prod && npm run build.scss",
+    "build.scss": "ts-node -P ./scripts/build/tsconfig.json ./scripts/build/build-css.ts",
+    "build.watch": "stencil build --watch",
     "lint": "eslint src/**/*{.ts,.tsx}",
     "start": "stencil build --dev --watch --serve",
-    "dev": "stencil build --watch",
     "test": "npm run test:spec && npm run test:ct",
     "test:spec": "stencil test --spec",
```
Comment (jul-lam) [inline packages/core/package.json:38]: test scipts sollten auch auf "." geändert werden.

**Review by jul-lam — CHANGES_REQUESTED** [review]

File: packages/storybook-docs/.storybook/preview.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  if (hasAdditionalTheme) {
+    additionalTheme['brand-dark'] = 'theme-brand-dark';
+    additionalTheme['brand-light'] = 'theme-brand-light';
+    // import('@siemens/ix-brand-theme/loader')
+    //   .then(({ defineCustomElements: defineAdditionalComponents }) =>
+    //     defineAdditionalComponents()
+    //   )
+    //   .catch(() => {
+    //     // fail silently
+    //   });
```
[the remark this answers is not in the document]
Comment (matthiashader) [reply packages/storybook-docs/.storybook/preview.ts]: has been removed, as a plugin takes care of it

**Review by jul-lam — DISMISSED** [review]

**Review by matthiashader — DISMISSED** [review]

**Pull request merged**